golem_messages/message.py

```diff
@@ -140,13 +139,23 @@ def raw(self):
         """Returns a raw copy of the message"""
         return self._raw[:]
 
-    def get_short_hash(self):
+    def get_short_hash(self, payload=None):
         """Return short message representation for signature
         :return bytes: sha1(TYPE, timestamp, encrypted, payload)
         """
+        if payload is None:
+            payload = serializer.dumps(self.slots())
         sha = hashlib.sha1()
-        sha.update(self.serialize_header())
-        sha.update(self._payload or b'')
+
+        # We can't use self.serialize_header() because it includes
+        # self.encrypted. And nested messages are decrypted, but they
+        # still need to have a valid signature.
+        # SEE: test_serializer.MessageTestCase.test_message_sig()
+        hash_header = serializer.dumps(

```

Comment:
`serialize_header` can be a `classmethod`, taking `timestamp` and `encrypted` as arguments.

Reply:
This would mean storing original `encrypted` value when serializing nested message with cbor. Sounds overcomplicated to me.